controllers/storagecluster/storagecluster_controller.go

```diff
@@ -144,28 +150,77 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
-	builder := ctrl.NewControllerManagedBy(mgr).
+	noobaaIgnoreTimeUpdatePredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*nbv1.NooBaa)
+			newObj := e.ObjectNew.(*nbv1.NooBaa)
+
+			diff := cmp.Diff(
+				oldObj, newObj,
+				cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
+				cmpopts.IgnoreFields(conditionsv1.Condition{}, "LastHeartbeatTime", "LastTransitionTime"),
```

Comment:
quick check, are ignored fields directly part of Condition struct? does this `cmpopts` recurse into sub types?

Reply:
Yes i have tested it, It works as expected.

Reply:
You can try it out.

```
package main

import (
	"fmt"

	"github.com/google/go-cmp/cmp"
	"github.com/google/go-cmp/cmp/cmpopts"
	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

type NooBaa struct {
	metav1.ObjectMeta `json:"metadata,omitempty"`
	Spec              NoobaaSpec   `json:"spec,omitempty"`
	Status            NooBaaStatus `json:"status,omitempty"`
}

type NoobaaSpec struct {
}

type NooBaaStatus struct {
	Conditions []conditionsv1.Condition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
}

func main() {
	n1 := &NooBaa{
		ObjectMeta: metav1.ObjectMeta{
			Name:            "n1",
			ResourceVersion: "1",
		},
		Status: NooBaaStatus{
			Conditions: []conditionsv1.Condition{
				{
					LastHeartbeatTime:  metav1.Now(),
					LastTransitionTime: metav1.Now(),
					Message:            "Hello1",
				},
			},
		},
	}

	n2 := &NooBaa{
		ObjectMeta: metav1.ObjectMeta{
			Name:            "n2",
			ResourceVersion: "2",
		},
		Status: NooBaaStatus{
			Conditions: []conditionsv1.Condition{
				{
					LastHeartbeatTime:  metav1.Now(),
					LastTransitionTime: metav1.Now(),
					Message:            "Hello2",
				},
			},
		},
	}

	diff := cmp.Diff(
		n1, n2,
		cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
		cmpopts.IgnoreFields(conditionsv1.Condition{}, "LastHeartbeatTime", "LastTransitionTime"),
	)
	if diff != "" {
		fmt.Println(diff)
	}
}

```